### ISSUE TITLE
manage_group not working

### ISSUE THREAD

**Issue by gothicx** [deric/puppet-accounts#46]
Hi. I've the following hiera structure, but it doesn't create the accounts / groups specified.

```
# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.2 (Maipo)
# puppet -V
4.4.0
```

``` yaml
accounts::user_defaults:
  purge_ssh_keys: false
accounts::users:
  account1:
    manage_group: false
    uid: 1417
    groups:
      - staff
    comment: 'test'
accounts::groups:
  staff:
    gid: 3000
```

```
Error: Could not create user account1: 
Execution of '/sbin/useradd -c test -g 1417 -d /home/account1-s /bin/bash -u 1417 -m account1' 
returned 6: useradd: group '1417' does not exist
```

It's using the uid as the gid.

If I remove the `manage_group = false` from the users, it does create the staff account., but I've others groups which it doesn't create as it's trying to create the users before the groups.

**Comment by gothicx** [comment]
If I remove the manage_group = false from the users, it does create the staff account., but I've others groups which it doesn't create.

**Comment by mobilesfinks** [comment]
Confirm issue
```
accounts::users:
    defaults:
      shell: /bin/bash
    manage: true
    users:
      dalp:
        uid: 1005
        comment: dalp
        groups:
        - users
      deployer:
        uid: 1010
        comment: dalp

```

> Error: /Stage[main]/Accounts::Users/Accounts::User[dalp]/User[dalp]/ensure: change from absent to present failed: Could not create user dalp: Execution of '/usr/sbin/useradd -c dalp -g dalp -d /home/dalp -s /bin/bash -u 1005 -m dalp' returned 6: useradd: group 'dalp' does not exist

> Error: /Stage[main]/Accounts::Users/Accounts::User[deployer]/User[deployer]/ensure: change from absent to present failed: Could not create user deployer: Execution of '/usr/sbin/useradd -c dalp -g deployer -d /home/deployer -s /bin/bash -u 1010 -m deployer' returned 6: useradd: group 'deployer' does not exist

**Comment by deric** [comment]
Doesn't seem to be just RedHat issue, same problem on Debian as well. Looks like duplicate of #41.
Here's a [minimal script](https://gist.github.com/deric/f48dd53ef7c16b56d442c8d9b6ccdd62) to reproduce the issue in clean environment.

**Comment by deric** [comment]
Should be fixed in v1.4.2
